apps/frontend/src/modules/surveys/SurveyProvider.tsx

```diff
@@ -73,20 +86,61 @@ export const SurveyProvider = ({ children }: SurveyProviderProp) => {
     const [deleteSurveyMutation, { loading: pendingDeletion }] =
         useSurveyDeleteMutation();
 
+    const [getSurvey, { loading: loadingSurvey, data: survey }] =
+        useGetSurveyLazyQuery({
+            variables: {
+                slug: surveySlug,
+            },
+        });
+
     const {
-        data: survey,
-        loading: loadingSurvey,
         refetch,
-    } = useGetSurveyQuery({
+        fetchMore,
+        data: surveyList,
+        loading: surveyListLoading,
+    } = useGetSurveyListQuery({
         variables: {
-            slug: surveySlug,
+            first: SURVEYS_PER_PAGE,
+            sortBy: {
+                field: SurveySortField.Name,
+                direction: OrderDirection.Desc,
+            },
+        },
+        context: {
+            headers: {
+                Project: workspace?.project.id,
+            },
         },
-        skip: !surveySlug,
+        notifyOnNetworkStatusChange: true,
     });
 
     const surveyName = survey?.survey?.name ?? "";
     const surveyId = survey?.survey?.id ?? "";
 
+    const totalCount = surveyList?.surveys?.totalCount;
+    const pageInfo = surveyList?.surveys?.pageInfo;
+
+    const transformedSurveyList = surveyList?.surveys?.edges
+        ?.map((edge) => {
+            return {
+                id: edge?.node?.id,
+                slug: edge?.node?.slug,
+                status: edge?.node?.status,
+                createdAt: edge?.node?.createdAt,
+                name: edge?.node?.name ? edge?.node?.name : "Untitled survey",
+                creator: {
+                    email: edge?.node?.creator.email,
+                    fullName: `${edge?.node?.creator.firstName} ${edge?.node?.creator.lastName}`,
+                },
+            };
+        })
+        .sort((a, b) => {
```

Comment:
@kaf-lamed-beyt Why are you sorting locally? All form of sorting should be from the backend.

Reply:
Yes. I added this as a placeholder until I'm able to do so.